components/contribution-flow/StepProfileLoggedInForm.js

```diff
@@ -35,8 +35,15 @@ const getProfileInfo = (stepProfile, profiles) => {
 };
 
 const StepProfileLoggedInForm = ({ profiles, onChange, collective, tier, data, stepDetails }) => {
-  const profileInfo = getProfileInfo(data, profiles);
-  const isContributingFromSameHost = data?.host?.id === collective.host.legacyId;
+  const profileInfo = React.useMemo(() => getProfileInfo(data, profiles), [data, profiles]);
+  const requiredInformation = React.useMemo(
+    () => getRequiredInformation(data, stepDetails, collective, profiles, tier),
+    [data, stepDetails, profiles, collective, tier],
+  );
+  const isContributingFromSameHost = React.useMemo(
+    () => data?.host?.id === collective.host.legacyId,
+    [data, collective],
+  );
```

Comment:
The result is a boolean, and a comparison between two integers is likely cheaper or around the same cost as comparing two references, so this memo won't improve performance - it will have to check if `data` and `collective` have changed since the last render.